mlir/lib/Dialect/Rock/Transforms/SugarToLoops.cpp

```diff
@@ -241,7 +247,24 @@ struct TransformingForRewritePattern
         cloneMap.map(validities[i], isValid);
       }
     }
-
+    // One-iteration loops should be inlined, do so here.
+    if (loops.empty()) {
+      auto yieldOp = cast<rock::YieldOp>(op.getBody()->getTerminator());
+      b.replaceAllUsesWith(op.getResults(), yieldOp.getOperands());
+      SmallVector<Value> blockArgValues;
+      for (auto [blockArg, initValue] :
+           llvm::zip(op.getIterArgs(), op.getIterInits())) {
+        cloneMap.map(blockArg, initValue);
+      }
+      blockArgValues.reserve(op.getBody()->getNumArguments());
+      for (auto blockArg : op.getBody()->getArguments())
+        blockArgValues.push_back(cloneMap.lookup(blockArg));
```

Comment:
Wouldn't this be the same as IterInits?

Reply:
No, the block arguments include things like the lower indices and the validity